src/compile/data/aggregate.ts

```diff
@@ -117,7 +117,8 @@ export class AggregateNode extends DataFlowNode {
         } else {
           meas[s.field] = meas[s.field] || {};
           meas[s.field][s.aggregate] = true;
-          as[s.field] = s.as;
+          as[s.field] = as[s.field] || {};
```

Comment:
Remove `as` -- make `meas` a `Dict<Dict<string>>` and always include the correct as. 
(For encoding's aggregate, call `field(...)` with the right params.